commit/chainfee/observation.go

```diff
@@ -31,6 +32,10 @@ func (p *processor) Observation(
 ) (Observation, error) {
 	lggr := logutil.WithContextValues(ctx, p.lggr)
 
+	if invalidateCache, ok := ctx.Value(consts.InvalidateCacheKey).(bool); ok && invalidateCache {
+		p.obs.invalidateCaches(ctx, lggr)
```

Comment:
This invalidates the cache but we can't be sure we have a viable reading by the time we observe, right? Would it make more sense to bust + read in a blocking way?